### PR TITLE
Fixes #5385. Add Markdown.RenderToAnsi() API for headless ANSI rendering

### LINKED ISSUE #5385: Add Markdown-to-ANSI rendering API on Markdown view
## Summary

The `Markdown` view can render markdown content interactively in the TUI, but there's no public API to render markdown content to ANSI escape sequences for stdout output (print/cat mode).

## Use Case

CLI tools built on Terminal.Gui (e.g. `clet`) need to support a `--cat` mode that renders markdown to the terminal without launching the interactive TUI. Currently this requires a workaround: creating a headless ANSI driver, initializing a `Markdown` view off-screen, calling `Draw()`, and extracting the ANSI output via `Driver.ToAnsi()`.

This pattern is brittle, requires internal knowledge of the driver lifecycle, and produces unnecessary allocations (full screen buffer for a print operation).

## Proposed API

A static or instance method on `Markdown` (or a companion utility class) that takes markdown text and returns an ANSI string:

```csharp
// Option A: static utility
string ansi = Markdown.RenderToAnsi (markdownText, width: 80);

// Option B: instance method (reuses existing view config like SyntaxHighlighter)
Markdown md = new () { SyntaxHighlighter = new TextMateSyntaxHighlighter () };
string ansi = md.RenderToAnsi (markdownText, width: Console.WindowWidth);
```

The method should:
- Accept a target width (defaults to 80 or Console.WindowWidth)
- Apply syntax highlighting if a `SyntaxHighlighter` is configured
- Return the rendered ANSI string without requiring `Application.Init()` or a running driver
- Handle content height automatically (render full document, not just viewport)

## Current Workaround

```csharp
Environment.SetEnvironmentVariable ("DisableRealDriverIO", "1");
IApplication app = Application.Create ();
app.Init (DriverRegistry.Names.ANSI);
app.Driver?.SetScreenSize (width, height);

Markdown markdownView = new () { Text = markdown, Width = Dim.Fill (), Height = Dim.Fill () };
markdownView.SetRelativeLayout (app.Screen.Size);
markdownView.Layout ();

int contentHeight = markdownView.GetContentHeight ();
app.Driver?.SetScreenSize (width, contentHeight);
markdownView.Frame = app.Screen with { X = 0, Y = 0 };
markdownView.Layout ();
app.Driver?.ClearContents ();
markdownView.Draw ();

string rendered = app.Driver?.ToAnsi () ?? string.Empty;
app.Dispose ();
```

This is ~20 lines of boilerplate that every consumer would need to duplicate.

## Context

This is tracked as a dependency for the `Terminal.Gui.Cli` library (see [terminal-gui-cli spec](https://github.com/gui-cs/clet/blob/develop/specs/terminal-gui-cli.md) section 5.7 MarkdownRenderer utility). The library currently defers this feature pending upstream support.

### PR DESCRIPTION
**Pull request by tig**
## Summary

Adds a public `RenderToAnsi` instance method on the `Markdown` view that renders markdown content to ANSI escape sequences without requiring `Application.Init()` or an interactive TUI session.

## API

```csharp
// Instance method - reuses view configuration (SyntaxHighlighter, Pipeline, etc.)
Markdown md = new () { SyntaxHighlighter = new TextMateSyntaxHighlighter () };
string ansi = md.RenderToAnsi (markdownText, width: Console.WindowWidth);
```

### Parameters
- `markdown` (optional) — Text to render. If `null`, uses the current `Text` property.
- `width` (optional) — Target column width for word-wrapping. Defaults to 80.

## How it works

The method encapsulates the ~20-line headless driver workaround described in the issue into a single call:

1. Sets `DisableRealDriverIO=1` to prevent real terminal I/O
2. Creates a temporary `IApplication` with the ANSI driver
3. Creates a fresh `Markdown` view copying configuration from the calling instance
4. Performs layout to determine content height
5. Resizes the buffer to the full document height and draws
6. Extracts and returns the ANSI output via `Driver.ToAnsi()`
7. Cleans up the temporary application

## Configuration properties respected

- `SyntaxHighlighter` — Syntax highlighting for code blocks
- `MarkdownPipeline` — Custom Markdig pipeline
- `UseThemeBackground` — Theme background colors
- `ShowHeadingPrefix` — Whether to show `#` prefixes

## Testing

- 9 new unit tests covering basic rendering, width wrapping, syntax highlighting, edge cases, and non-mutation of the calling instance
- All 374 existing Markdown tests continue to pass

## Closes #5385

### CODE REVIEW

File: Terminal.Gui/Views/Markdown/MarkdownView.Ansi.cs [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+        // Suppress real terminal I/O for the headless driver
+        string? previousValue = Environment.GetEnvironmentVariable ("DisableRealDriverIO");
+        Environment.SetEnvironmentVariable ("DisableRealDriverIO", "1");
+
+        try
+        {
+            IApplication app = Application.Create ().Init (DriverRegistry.Names.ANSI);
+            app.Driver!.SetScreenSize (width, width);
+
+            Markdown renderView = new ()
```
[the remark this answers is not in the document]
Comment (tig) [reply Terminal.Gui/Views/Markdown/MarkdownView.Ansi.cs]: Good catch -- switched both the IApplication and the temporary Markdown view to using declarations so disposal is guaranteed regardless of exceptions during layout, draw, or syntax highlighting.

File: Terminal.Gui/Views/Markdown/MarkdownView.Ansi.cs [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+
+        if (string.IsNullOrEmpty (text))
+        {
+            return string.Empty;
+        }
+
+        // Suppress real terminal I/O for the headless driver
+        string? previousValue = Environment.GetEnvironmentVariable ("DisableRealDriverIO");
+        Environment.SetEnvironmentVariable ("DisableRealDriverIO", "1");
+
```
[the remark this answers is not in the document]
Comment (tig) [reply Terminal.Gui/Views/Markdown/MarkdownView.Ansi.cs]: I believe there's an IDriver property that could be used insted? or maybe it's on IApplication. if not, there should be and a new TG Issue should be filed to add it.

File: Terminal.Gui/Views/Markdown/MarkdownView.Ansi.cs [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+    ///         <see cref="ShowHeadingPrefix"/>, and <see cref="ShowCopyButtons"/> — are copied to the
+    ///         temporary view used for rendering.
+    ///     </para>
+    /// </remarks>
+    public string RenderToAnsi (string? markdown = null, int width = 80)
+    {
+        if (width < MIN_WRAP_WIDTH)
+        {
+            width = MIN_WRAP_WIDTH;
+        }
```
[the remark this answers is not in the document]
Comment (tig) [reply Terminal.Gui/Views/Markdown/MarkdownView.Ansi.cs:42]: Accepted. Added a MAX_RENDER_WIDTH constant (4096) that clamps the width parameter to prevent excessive allocations.

File: Terminal.Gui/Views/Markdown/MarkdownView.Ansi.cs (hunk 2 above)
[the remark this answers is not in the document]
Comment (tig) [reply Terminal.Gui/Views/Markdown/MarkdownView.Ansi.cs]: Accepted. Wrapped the env var set/restore in a static lock so concurrent calls cannot race.

File: Terminal.Gui/Views/Markdown/MarkdownView.Ansi.cs [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+        {
+            using IApplication app = Application.Create ().Init (DriverRegistry.Names.ANSI);
+            app.Driver!.SetScreenSize (width, width);
+
+            using Markdown renderView = new ()
+            {
+                Text = text,
+                Width = Dim.Fill (),
+                Height = Dim.Fill (),
+                SyntaxHighlighter = SyntaxHighlighter,
```
[the remark this answers is not in the document]
Comment (tig) [reply Terminal.Gui/Views/Markdown/MarkdownView.Ansi.cs]: This was already addressed in the previous commit (dba6ef93) which switched to using declarations for both objects. No additional change needed.

File: Terminal.Gui/Views/Markdown/MarkdownView.Ansi.cs [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+        }
+
+        // Suppress real terminal I/O for the headless driver
+        string? previousValue = Environment.GetEnvironmentVariable ("DisableRealDriverIO");
+        Environment.SetEnvironmentVariable ("DisableRealDriverIO", "1");
+
+        try
+        {
+            using IApplication app = Application.Create ().Init (DriverRegistry.Names.ANSI);
+            app.Driver!.SetScreenSize (width, width);
```
[the remark this answers is not in the document]
Comment (tig) [reply Terminal.Gui/Views/Markdown/MarkdownView.Ansi.cs]: Accepted. Changed initial height from width to 1 for the first layout pass -- only the full content height allocation is needed for the final draw.

File: Terminal.Gui/Views/Markdown/MarkdownView.Ansi.cs [hunk 6]
```diff
@@ -0,0 +1,92 @@
...
+    ///     <para>
+    ///         This method does not require <see cref="Application"/> to be initialized. It creates a
+    ///         temporary headless ANSI driver internally, performs layout and drawing into an off-screen
+    ///         buffer, and returns the ANSI representation.
+    ///     </para>
+    ///     <para>
+    ///         Configuration properties set on this instance — <see cref="SyntaxHighlighter"/>,
+    ///         <see cref="MarkdownPipeline"/>, <see cref="UseThemeBackground"/>,
+    ///         <see cref="ShowHeadingPrefix"/>, and <see cref="ShowCopyButtons"/> — are copied to the
+    ///         temporary view used for rendering.
```
[the remark this answers is not in the document]
Comment (tig) [reply Terminal.Gui/Views/Markdown/MarkdownView.Ansi.cs]: Accepted. Updated the XML docs to clarify that ShowCopyButtons is always disabled for ANSI output since copy buttons are interactive-only controls.

File: Terminal.Gui/Views/Markdown/MarkdownView.Ansi.cs (hunk 2 above)
[the remark this answers is not in the document]
Comment (tig) [reply Terminal.Gui/Views/Markdown/MarkdownView.Ansi.cs]: I checked -- there is no existing property on IDriver or IApplication to force headless mode. The env var is checked in Driver.IsAttachedToTerminal() and TerminalDevice. Adding a proper API (e.g. a static Driver.ForceHeadless flag or an IApplication init option) would be a separate feature. Want me to file an issue for that? For now the static lock keeps this method safe against concurrent use.

**Review by BDisp — APPROVED** [review]
